src/connectors/range/connectRange.js

```diff
@@ -305,43 +300,39 @@ export default function connectRange(renderFn, unmountFn = noop) {
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        const value = uiState && uiState.range && uiState.range[attribute];
+        let withFacetConfiguration = searchParameters.setQueryParameters({
+          disjunctiveFacets: [attribute],
+          numericRefinements: {
+            ...searchParameters.numericRefinements,
+            [attribute]: {},
+          },
+        });
+
+        const value = uiState.range && uiState.range[attribute];
 
         if (!value || value.indexOf(':') === -1) {
-          return searchParameters;
+          return withFacetConfiguration;
         }
 
-        const {
-          '>=': previousMin = [NaN],
-          '<=': previousMax = [NaN],
-        } = searchParameters.getNumericRefinements(attribute);
-        let clearedParams = searchParameters.clearRefinements(attribute);
         const [lowerBound, upperBound] = value.split(':').map(parseFloat);
 
-        if (
-          previousMin.includes(lowerBound) &&
-          previousMax.includes(upperBound)
-        ) {
-          return searchParameters;
-        }
-
         if (isFiniteNumber(lowerBound)) {
-          clearedParams = clearedParams.addNumericRefinement(
+          withFacetConfiguration = withFacetConfiguration.addNumericRefinement(
```

Comment:
I would have use a variable for each block, it's easier to follow IMO: `withLowerBound`, `withUpperBrand`.

Reply:
Here we need to mutate them so I'm not sure it makes them more readable. Besides, it creates a naming burden.